src/optimize.js

```diff
@@ -0,0 +1,78 @@
+import once from 'lodash/once';
+import { Experiment } from './models/experiment';
+
+class Optimize implements Experiment {
+  listeners: Array<Function> = [];
+  eventData: any;
+
+  constructor(experimentName, controlProps) {
+    this.experimentName = experimentName;
+
+    if (controlProps) {
+      this.controlProps = controlProps;
+    }
+  }
+
+  activate = () => {
+    this.addEventListener();
+
+    // Trigger an experiment execute is a async operation hence we need
+    // to fire an event from experiment to trigger the rendering from Google Optimize.
+    // This way we are ensuring that Optimize had time to execute the script in the exeriment.
+    console.log(`${this.experimentName}.activate`);
```

Comment:
is this needed?

Reply:
If we need to do something on activation - its better to have a `onActivation` callback and not have a hard coded log here